src/menu.js

```diff
@@ -103,31 +103,50 @@ class Menu extends Component {
   removeStatus() {
     ReactDOM.unmountComponentAtNode(this._statusNode)
     document.body.removeChild(this._statusNode)
+    this._statusNode = null
   }
 
   renderStatus = () => {
-    ReactDOM.render(
-      <MenuStatus
-        getA11yStatusMessage={this.props.getA11yStatusMessage}
-        getInputValue={this.autocomplete.getInputValue}
-        getItemFromIndex={this.getItemFromIndex}
-        highlightedIndex={this.state.highlightedIndex}
-        inputValue={this.autocomplete.state.inputValue}
-        resultCount={this.items.length}
-      />,
-      this._statusNode,
-    )
+    this._statusNode &&
+      ReactDOM.render(
+        <MenuStatus
+          getA11yStatusMessage={this.props.getA11yStatusMessage}
+          getInputValue={this.autocomplete.getInputValue}
+          getItemFromIndex={this.getItemFromIndex}
+          highlightedIndex={this.state.highlightedIndex}
+          inputValue={this.autocomplete.state.inputValue}
+          resultCount={this.items.length}
+        />,
+        this._statusNode,
+      )
   }
 
   getChildContext() {
     return {[MENU_CONTEXT]: this}
   }
 
-  componentDidMount() {
-    this.autocomplete.setMenu(this)
-    this.autocomplete.emitter.on('menu:open', this.setDefaultHighlightedIndex)
+  handleMenuOpen = () => {
+    const lastActiveElement = document.activeElement
+    this.returnFocus = () => {
+      // only return focus if a non-focusable item was selected when closing
+      if (document.activeElement === document.body) {
```

Comment:
Yes as long as you don't click another focusable item this will return true. So imagine you click another menu button, you would want that to be focused and not return focus to the button that opened it. I hope that makes sense.